### PR TITLE
Feature/create skeleton for data sync rules

### LINKED ISSUE #166: Create Skeleton for Data Sync Rule
- [x] Step 1: General Info (Integrate from metadata sync rules)
- [x] Step 2: Selection of DEs (empty)
- [x] Step 3: Selection of Org Units (empty)
- [x] Step 4: Selection of Period (empty)
- [x] Step 5: Selection of Category Option (empty)
- [x] Step 6: Selection of Instances (Integrate from metadata sync rules)
- [x] Step7: Scheduling(Integrate from metadata sync rules)
- [x] Step 8: Summary(Integrate from metadata sync rules)

### PR DESCRIPTION
**Pull request by xurxodev**
### :pushpin: References

* **Issue:** Closes #166 

### :memo: Implementation

- I have renamed current SyncRulesWizard to MetadataSyncRulesWizard
- I have created new folders under rules-creation-page: common, data and metadata. 
- I have moved common steps between data and metadata to common folder
- I have created DataSyncRulesWizard with new empty steps 
- I have created commonStepsBaseInfo with base info for common steps in data and metadata to avoid duplicate code.
- I have extracted common code of sync rule wizards creating a new GenericSyncRulesWizard component and this one is used from MetadataSyncRulesWizard and DataSyncRulesWizard by composition.

### :art: Screenshots

DataSyncRulesWizard

### :fire: Is there anything the reviewer should know to test it?

- On this PR  syncRule model is used for data and metadata rules wizards.  For data, the wizard tries to save using SyncRule in rules key of data store like in metadata but SyncRule model validations are fired before to save and metadataIds empty error is shown as a SnackBar

### :bookmark_tabs: Others

- To future PR
    - On this PR MetadataSyncRulesPage and DataSyncRulesPage have many duplicated code. I have added hardcoded empty results page to show in table of DataSyncRulesPage.  to avoid duplicate code, we have to think a solution, create generic and to use composition as other cases?

    - Currently, SyncRule model is used for data and metadata, we have to think if this is the approach to follow. If yes, modify SyncRule to support new steps in data sync rules, if not we need to extract SyncRule model from common steps.

### CODE REVIEW

**Comment by SferaDev** [comment]
Since our git flow is a bit more complex now with ``development-who`` and ``development-malaria``. Maybe it would be useful to have intermediary PR for fast-forwarding/merging upstream. I find it really useful to use the "Files changed" for code review but with our current flow it's hard to keep track of it.

About the PR I will try to take a look later today or tomorrow morning (worst case). Thanks!

**Comment by xurxodev** [comment]
@SferaDev @adrianq PR ready to review!

File: src/components/rules-list-page/DataSyncRulesPage.jsx [hunk 1]
```diff
@@ -372,6 +372,12 @@ class DataSyncRulesPage extends React.Component {
         );
     };
 
+    //TODO: for the moment show empty data sync rules
+    emptyDataSyncRules = (d2, filters, pagination) =>
+        new Promise((resolve, reject) => {
+            resolve({ objects: [], pager: { page: 1, pageCount: 1, total: 0 } });
+        });
```
[the remark this answers is not in the document]
Comment (xurxodev) [reply src/components/rules-list-page/DataSyncRulesPage.jsx]: ok thanks for the explanation

**Comment by adrianq** [comment]
Thanks @xurxodev It works like a charm! Something very minor, the sentence 'Importing data in instance {{instance}}' is used both for metadata and data when syncing. In your next PR, if it is not too difficult, show a different message depending if it is a metadata or a data sync.

**Pull request merged**